### DISCUSSION TITLE
Allowing users to add and edit shipping payment details

### DISCUSSION THREAD

**Discussion by flonUK** [swellstores/community#487] [Help] [answered]
It is not possible for users to add/update their personal details: shipping/payment details on the customer dashboard. Some have subscriptions in their accounts, but cannot add payment details. Any suggestions on how to fix will be greatly appreciated.

Thanks
<img width="1103" alt="Screenshot 2024-01-30 at 09 14 46" src="https://github.com/swellstores/community/assets/156616226/c4bfe829-3c8b-41db-8c03-aabbd6303938">

**Comment by logeshswell** [comment, marked as answer]
Yes, it is a gap in our [Horizon theme](https://github.com/swellstores/horizon). You can extend this project to add support for adding shipping and payment details as follows using GraphQL:
https://developers.swell.is/frontend-api/carts#update-cart-shipping-info
https://developers.swell.is/frontend-api/carts#update-cart-billing-info
